pull_request.py

```diff
@@ -220,6 +220,67 @@ def diff(self) -> unidiff.PatchSet:
         self._repo, self._pr_number))
     return unidiff.PatchSet(response.content.decode('utf-8'))
 
+  def get_new_bonuses_or_raise(self) -> List[Tuple[str, str, int]]:
+    # If this PR represents adding a new bonus files, return details
+    # about the files.  Otherwise raise an exception explaining how the diff
+    # doesn't qualify.
+    #
+    # Returns [(bonus_1_user, bonus_1_name, bonus_1_value),
+    #          (bonus_2_user, bonus_2_name, bonus_2_value), ... ]
+
+    diff = self.diff()
+    if diff.modified_files or diff.removed_files:
+      raise Exception('All file changes must be additions')
+
+    bonuses: List[Tuple[str, str, int]] = []
+
+    for added_file in diff.added_files:
+      s_players, points_user, s_bonuses, bonus_name =  added_file.path.split('/')
+      if s_players != 'players' or s_bonuses != 'bonuses':
+        raise Exception('Added file %s is not a bonus file' % added_file)
+
+      (diff_invocation_line, file_mode_line, _, removed_file_line,
+       added_file_line, patch_location_line, file_delta_line,
+       empty_line) = str(added_file).split('\n')
+
+      if diff_invocation_line != 'diff --git a/%s b/%s' % (
+          added_file.path, added_file.path):
+        raise Exception('Unexpected diff invocation: %s' % diff_invocation_line)
+
+      if file_mode_line != 'new file mode 100644':
+        raise Exception('File added with incorrect mode: %s' % file_mode_line)
+
+      if removed_file_line != '--- /dev/null':
+        raise Exception(
+          'Diff format makes no sense: added files should say they are from /dev/null')
+
+      if added_file_line != '+++ b/%s' % added_file.path:
+        raise Exception('Something wrong with file adding line: file is '
+                        '%s but got %s' % (added_file.path, added_file_line))
+
+      if patch_location_line != '@@ -0,0 +1,1 @@':
+        raise Exception('Patch location makes no sense: %s' %
+                        patch_location_line)
+
+      if empty_line:
+        raise Exception('Last line should be empty')
+
+      if file_delta_line.startswith('+'):
+        actual_file_delta = file_delta_line[1:]
+      else:
+        raise Exception('File delta missing initial + for addition: %s' %
+                        file_delta_line)
+
+      # If this isn't an int, then it raises and the PR isn't mergeable
+      points_change = int(actual_file_delta)
```

Comment:
I think this should be wrapped in a try/catch for a more explicit error, instead of just `ValueError: invalid literal for int() with base 10: 'jeffkaufman wins the game by doing something sneaky'`

Reply:
Yeah, I made a slight change there along those lines in #226

Reply:
Done in 729c5d9, though since this is in "should we merge" logic it's in the wrong part of the codebase for something sneaky like that